### PR TITLE
manifest: Update hal_nxp to add Wi-Fi driver

### PR DESCRIPTION
**Pull request by MaochenWang1**
Update hal_nxp to add Wi-Fi driver code.

### CODE REVIEW

**Review by bjarki-andreasen — DISMISSED** [review]

**Comment by bjarki-andreasen** [comment]
looks good, ping when its a sha :)

**Review by dleach02 — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
Updated this PR with the contents of #80082, as the WiFi driver was previously not building in CI (and did not build without the changes now applied to this PR)

File: modules/hostap/src/supp_main.h [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	!defined(CONFIG_EAP_PSK) && !defined(CONFIG_EAP_PAX) && \
 	!defined(CONFIG_EAP_SAKE) && !defined(CONFIG_EAP_GPSK) && \
 	!defined(CONFIG_EAP_PWD) && !defined(CONFIG_EAP_EKE) && \
-	!defined(CONFIG_EAP_IKEV2 && !defined(CONFIG_EAP_GTC)
+	!defined(CONFIG_EAP_IKEV2) && !defined(CONFIG_EAP_GTC)
```
Comment (krish2718) [inline modules/hostap/src/supp_main.h:18]: Also fixed by https://github.com/zephyrproject-rtos/zephyr/pull/79974/commits/de54470bac24e8f6d201105a12ff867fd38d383f

File: modules/hostap/src/supp_main.c [hunk 2]
```diff
@@ -98,7 +98,6 @@ static const struct wifi_mgmt_ops mgmt_ap_ops = {
 	.ap_enable = supplicant_ap_enable,
 	.ap_disable = supplicant_ap_disable,
 	.ap_sta_disconnect = supplicant_ap_sta_disconnect,
-	.ap_bandwidth = supplicant_ap_bandwidth,
```
Comment (krish2718) [inline modules/hostap/src/supp_main.c:101]: please make a separate commit for hostap, and I guess the missing definition `supplicant_ap_bandwidth` is what is incomplete? and this is an accidental change?

File: modules/hostap/src/supp_main.c [hunk 3]
```diff
@@ -94,7 +94,6 @@ DEFINE_WIFI_NM_INSTANCE(wifi_supplicant, &mgmt_ops);
 
 #ifdef CONFIG_WIFI_NM_HOSTAPD_AP
 static const struct wifi_mgmt_ops mgmt_ap_ops = {
-	.set_btwt = supplicant_set_btwt,
```
Comment (krish2718) [inline modules/hostap/src/supp_main.c:97]: same as AP bandwidth, an accidental change now being reverted?

File: samples/net/wifi/boards/rd_rw612_bga.conf [hunk 4]
```diff
@@ -75,33 +75,6 @@ CONFIG_NET_SOCKETS_SERVICE_THREAD_PRIO=3
 CONFIG_NET_TC_SKIP_FOR_HIGH_PRIO=y
 CONFIG_NET_CONTEXT_PRIORITY=y
 
-CONFIG_WIFI_NM_WPA_SUPPLICANT=y
```
Comment (krish2718) [inline samples/net/wifi/boards/rd_rw612_bga.conf:78]: but without this will Wi-Fi work on RW612?

File: west.yml [hunk 5]
```diff
@@ -198,7 +198,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 4597b16cfedf5553cb155151e65eb994d5d0ef25
+      revision: pull/454/head
```
Comment (krish2718) [inline west.yml:201]: is this the right PR? I don't see any Wi-Fi driver source files added to the HAL, just the CI flag which is linked to another manifest PR?

File: modules/hostap/src/supp_main.c (hunk 3 above)
Comment (MaochenWang1) [reply modules/hostap/src/supp_main.c:97]: this is caused by wrong upstream sequence from downstream

File: samples/net/wifi/boards/rd_rw612_bga.conf (hunk 4 above)
Comment (MaochenWang1) [reply samples/net/wifi/boards/rd_rw612_bga.conf]: remove this commit

File: west.yml (hunk 5 above)
Comment (MaochenWang1) [reply west.yml:201]: wifi driver already merged into hal_nxp, now this links to another PR to fix the build issue, which is caused by one hostap PR still under review.

Comment (dleach02) [reply west.yml:201]: @krish2718, we were in the process of merging things when we found that there was a build problem. So we added the appropriate fix on this side and a change so that CI will actually build this platform. We used a path that Nordic did for platforms that require a blob to do a "build only" kconfig so CI would at least catch the error.

Comment (krish2718) [reply west.yml:201]: Sure, I was confused by the title that `Wi-Fi driver supprot added to HAL` but in that PR didn't see those changes.

File: modules/hostap/src/supp_main.h (hunk 1 above)
Comment (krish2718) [reply modules/hostap/src/supp_main.h:18]: @jukkar how do you want deal with this? Merge this and abandon that PR or wait for it to be merged and rebase this?

**Review by krish2718 — DISMISSED** [review]

File: modules/hostap/src/supp_main.h (hunk 1 above)
Comment (dleach02) [reply modules/hostap/src/supp_main.h:18]: I think you should let the other PR still move forward. @jukkar PR is needed to add the test case going forward.

**Comment by MaochenWang1** [comment]
Hi @jukkar  @krish2718  please review again

**Review by krish2718 — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by dleach02** [comment]
@bjarki-andreasen it is a sha

**Review by bjarki-andreasen — DISMISSED** [review]

File: drivers/wifi/nxp/Kconfig.nxp [hunk 6]
```diff
@@ -20,6 +20,13 @@ module = WIFI_NXP
 config WIFI_MGMT_SCAN_CHAN_MAX_MANUAL
 	default 50
 
+config NXP_WIFI_BUILD_ONLY_MODE
+	bool "Build only mode (do not link wireless binary blob)"
```
Comment (krish2718) [inline drivers/wifi/nxp/Kconfig.nxp:24]:
```suggestion
	bool "Build only mode (do not link wireless firmware blob)"
```

**Review by krish2718 — DISMISSED** [review]

**Comment by dleach02** [comment]
CI failure is with a new test that merged between last night and resubmission of the PR with a SHA. Working to fix it now :-(

File: drivers/wifi/nxp/Kconfig.nxp (hunk 6 above)
Comment (danieldegrasse) [reply drivers/wifi/nxp/Kconfig.nxp]: Fixed, thanks

**Review by dleach02 — APPROVED** [review]

**Review by gannicus-gj — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Comment by MaochenWang1** [comment]
Hi @krish2718  @bjarki-andreasen pls review again

**Review by krish2718 — APPROVED** [review]
